### DISCUSSION TITLE
What video codecs are supported?

### DISCUSSION THREAD

**Discussion by dejawooh** [kean/Nuke#518] [Q&A]
Firstly, this is an awesome community. Thanks for building it. I am trying to use Nuke for videos in my iOS project. 
I looked at the Nuke Demo project and the video in that project plays just fine. But when I use a different video it fails with error "Failed to create an image from the image data" in TaskFetchDecodedImage class. On further debugging I narrowed down to the video codec being different between the videos used in Nuke Demo (H.264) Vs the video (H.264, AAC) I use. When checking the AVFileType.mp4.rawvalue for Nuke Demo video it shows as public.mpeg-4 while the one I use is  "video/mp4". I have several videos in my project with this codec and content-type.

I am using LazyImage to display video in a View.

Is there a way to extend support to various content type and video codecs?

**Comment by kean** [comment]
Hey, @dejawooh! The video support is still in a somewhat experimental state – which we should probably document somewhere. I'm not a video expert, and the demand is largely coming from the community. I know the set of currently supported formats is limited and I would appreciate contributions.

[`AssertType`](https://github.com/kean/Nuke/blob/63f10a8d97f9e9b06606131ac85298e859dfa1fa/Sources/Core/Decoding/AssetType.swift#L53) determines what formats are recognized by Nuke. It uses magic numbers, and I would've loved to replace it with something native, but I don't think there is anything available right now, especially not with the support for the older OS versions. The playback is done using `AVKit`, so whatever it supports should also be supported. It looks like it's just a matter of adding support for more formats.

**Reply by dejawooh** [reply]
Hi Kean, Thanks for incredibly fast response. Let me see if I can find a way to extend the video codec/format support without messing the existing code. I will share my solution here if I am able to solve. 

Thanks and much appreciated.
